### PR TITLE
SCOR-216: check ip from connection before return

### PR DESCRIPTION
**Pull request by 0lvin**

### CODE REVIEW

**Comment by kostya13** [comment]
Looks good +1

**Comment by yoramw** [comment]
I see you throw a non recoverable error in case the IP is not set. Shouldn't it be recoverable error if it is later on being set?

**Comment by GlaciErrDev** [comment]
+1

File: network_plugin/__init__.py [hunk 1]
```diff
@@ -258,7 +258,7 @@ def get_ondemand_public_ip(vca_client, gateway, ctx):
         new_ip = list(available_ips)[0]
         ctx.logger.info("Public IP {0} was reused.".format(new_ip))
         return new_ip
-    for i in range(5):
+    for i in xrange(5):
```
Comment (kostya13) [inline network_plugin/__init__.py]: Please use variable, instead number.

File: network_plugin/__init__.py [hunk 2]
```diff
@@ -325,12 +325,32 @@ def get_gateway(vca_client, gateway_name):
...
         retry_after=GATEWAY_TIMEOUT)
 
 
+def retry_operation(func):
+    """
+        retry ten times to run funtion with sleep several second between
+        retry
+    """
+    def f(*args, **kwargs):
+        for _ in xrange(10):
```
Comment (kostya13) [inline network_plugin/__init__.py]: Please use variable, instead number.

File: network_plugin/__init__.py [hunk 3]
```diff
@@ -350,7 +370,8 @@ def save_ssh_parameters(ctx, port, ip):
 
 
 def wait_for_gateway(vca_client, gateway_name, ctx):
-    for i in range(10):
+    """try ten times to wait 10 seconds for gateway"""
+    for i in xrange(10):
```
Comment (kostya13) [inline network_plugin/__init__.py]: Please use variable, instead number.

File: network_plugin/__init__.py [hunk 4]
```diff
@@ -325,12 +325,32 @@ def get_gateway(vca_client, gateway_name):
...
 def set_retry(ctx):
+    """
+        set retry on cloudify level
+    """
     return ctx.operation.retry(
         message='Waiting for gateway.',
         retry_after=GATEWAY_TIMEOUT)
 
 
+def retry_operation(func):
```
Comment (kostya13) [inline network_plugin/__init__.py]: Whw did you not use OperationRetry?

File: network_plugin/__init__.py [hunk 5]
```diff
@@ -37,11 +37,15 @@
 BUSY_MESSAGE = "is busy completing an operation"
 
 GATEWAY_TIMEOUT = 30
+# try ten times before fail
+RETRY_COUNT = 10
+# sleep 10 seconds before retry
```
Comment (kostya13) [inline network_plugin/__init__.py]: Don't use value here. The value user can see in assignment.

File: network_plugin/__init__.py [hunk 6]
```diff
@@ -37,11 +37,15 @@
 BUSY_MESSAGE = "is busy completing an operation"
 
 GATEWAY_TIMEOUT = 30
+# try ten times before fail
```
Comment (kostya13) [inline network_plugin/__init__.py]: Don't value here. The value user can see in assignment.

File: server_plugin/server.py [hunk 7]
```diff
@@ -267,6 +274,21 @@ def delete(vca_client, **kwargs):
     del ctx.instance.runtime_properties[VCLOUD_VAPP_NAME]
 
 
+def _all_primary_connections_have_ip(vapp):
```
Comment (kostya13) [inline server_plugin/server.py]: VM can have only one primary connection.
Therefore function must named: "_is_primary_connection_has_ip"

File: server_plugin/server.py [hunk 8]
```diff
@@ -267,6 +274,21 @@ def delete(vca_client, **kwargs):
...
+    """Return True in case when all primary interfaces have some ip"""
```
Comment (kostya13) [inline server_plugin/server.py]: "Return True in case when  primary interface has some ip"

**Comment by kostya13** [comment]
Looks good +1

**Pull request merged**